src/crimson/osd/shard_services.cc

```diff
@@ -161,23 +161,12 @@ void ShardServices::remove_want_pg_temp(pg_t pgid)
   pg_temp_pending.erase(pgid);
 }
 
-void ShardServices::_sent_pg_temp()
-{
-#ifdef HAVE_STDLIB_MAP_SPLICING
-  pg_temp_pending.merge(pg_temp_wanted);
-#else
-  pg_temp_pending.insert(make_move_iterator(begin(pg_temp_wanted)),
-			 make_move_iterator(end(pg_temp_wanted)));
-#endif
-  pg_temp_wanted.clear();
-}
-
 void ShardServices::requeue_pg_temp()
 {
   unsigned old_wanted = pg_temp_wanted.size();
   unsigned old_pending = pg_temp_pending.size();
-  _sent_pg_temp();
-  pg_temp_wanted.swap(pg_temp_pending);
+  pg_temp_wanted.merge(pg_temp_pending);
+  pg_temp_pending.clear();
```

Comment:
I thought merge will clear the parameter map....

Reply:
right. thought it could caused by `forced` pg temp. it it never happens to `pg_temp_wanted.merge()`. will add a commit to drop it to make sure it's removed explicitly.

Reply:
after a second thought, i think "forced" pg temp should override the ones in `pg_temp_pending` with the same pg id, so we should not remove `pg_temp_pending.clear()` call.

the "forced" pg temp is not enqueued by crimson, but it is used by classic osd. before we have good reason to drop "forced" pg temp support, i'm inclined to keep it.

Reply:
I agree:-)